common/connect/src/main/java/org/csgroup/csc/s2/repository/WebdavRepository.java

```diff
@@ -23,6 +27,13 @@ public WebdavRepository(WebdavRepositoryConfiguration config)
             throws KeyManagementException, NoSuchAlgorithmException, KeyStoreException, CertificateException,
             IOException {
         this.baseUrl = config.getWebdavServiceURL();
+        var path = this.baseUrl.getPath();
+        if (! path.endsWith(TRAILING_SLASH)) {
+            this.baseUrl = new URL(this.baseUrl.toString().replaceFirst(path, path + TRAILING_SLASH));
```

Comment:
I do not understand why use the `replaceFirst`

```suggestion
            this.baseUrl = new URL(this.baseUrl.toString() + TRAILING_SLASH);
```